src/main/java/edu/escuelaing/ieti/matchmaking/services/impl/TeamServiceImpl.java

```diff
@@ -14,31 +15,44 @@
 @Service
 public class TeamServiceImpl implements TeamService {
 	@Autowired
-	private TeamRepository inMemoryRepository;
+	private TeamRepository teamRepository;
 
 	@Override
 	public Team create(Team team) throws EntityExistsException {
-		return inMemoryRepository.create(team);
+		if (teamRepository.existsById(team.getTeamId())){
+            throw new EntityExistsException(Team.class, "Team Id", team.getTeamId());
+        }
+        return teamRepository.save(team);
 	}
 
 	@Override
-	public void remove(String teamId) throws EntityNotFoundException {
-		inMemoryRepository.remove(teamId);
+	public void remove(Team team) throws EntityNotFoundException {
+		if (teamRepository.existsById(team.getTeamId())){
+            throw new EntityNotFoundException(Team.class, "Team Id", team.getTeamId());
+        }
+		teamRepository.delete(team);
 	}
 
 	@Override
 	public Team update(Team team) throws EntityNotFoundException {
-		return inMemoryRepository.update(team);
+		if (teamRepository.existsById(team.getTeamId())){
+            throw new EntityNotFoundException(Team.class, "Team Id", team.getTeamId());
+        }
+		return teamRepository.save(team);
 	}
 
 	@Override
 	public Team getTeamById(String teamId) throws EntityNotFoundException {
-		return inMemoryRepository.getTeamById(teamId);
+		 Optional<Team> optionalTeam = teamRepository.findById(teamId);
+	        if (!optionalTeam.isPresent()){
```

Comment:
Check an `Optional` for presence inside a conditional statement is not a good practice. Take a look into the *effective java 3rd edition* book for proper patterns of `Optional`.